File: Task/Averages-Arithmetic-mean/Euphoria/averages-arithmetic-mean.ex
function mean(sequence s)
  atom sum
  if length(s) = 0 then
    return 0
  else
    sum = 0
    for i = 1 to length(s) do
      sum += s[i]
    end for
    return sum/length(s)
  end if
end function

sequence test
test = {1.0, 2.0, 5.0, -5.0, 9.5, 3.14159}
? mean(test)

File: Task/Averages-Arithmetic-mean/Zen-C/averages-arithmetic-mean.c
import "std/vec.zc"

fn mean(v: Vec<f64>*) -> f64 {
    if v.is_empty() {
        return 0.0;
    }

    let sum: f64 = 0.0;
    for x in *v {
        sum += x;
    }

    return sum / (f64)v.length();
}

fn main() {
    let numbers = Vec<f64>::new();
    numbers.push(10.0);
    numbers.push(20.0);
    numbers.push(30.0);
    numbers.push(40.0);

    let m = mean(&numbers);
    println "Numbers: 10, 20, 30, 40";
    println "Mean: {m}";

    let empty = Vec<f64>::new();
    let m_empty = mean(&empty);
    println "Empty mean: {m_empty}";
}
